specs/fault-proof/stage-one/optimism-portal.md

```diff
@@ -0,0 +1,168 @@
+[withdrawals](../../protocol/withdrawals.md). Because of the probabilistic validity of games as discussed in
+[AnchorStateRegistry](./anchor-state-registry.md), we can't immediately finalize withdrawals. Instead, we must wait for
+both a [dispute game finality delay](./anchor-state-registry.md#dispute-game-finality-delay) and a **proof maturity
+delay** to pass before we can finalize a withdrawal. Meanwhile, our assumptions about the fault proof system do work
+such that by the time the withdrawal is finalized, we're confident the withdrawal is correct.
+
+## Definitions
+
+### Authorized input
+
+An input for which there is social consensus, i.e. coming from governance.
+
+### Proven withdrawal
+
+A **proven withdrawal** is a withdrawal that is maybe valid, because it's been proven using a **maybe valid game**.
+However, because we don't have full confidence in the game's validity, we can't yet finalize the withdrawal.
+
+### Finalized withdrawal
+
+A **finalized withdrawal** is a withdrawal transaction that has been proven against a game that we now know is
+**valid**, and has waited the **proof maturity delay**.
+
+### Proof maturity delay
+
+The **proof maturity delay** is time that must elapse between a withdrawal being proven and it being finalized.
+
+## Assumptions
+
+### aASR-001: AnchorStateRegistry correctly distinguishes maybe valid games
+
+We assume that the AnchorStateRegistry correctly reports whether a game is a [**maybe valid
+game**](./anchor-state-registry.md#maybe-valid-game).
+
+#### Impact
+
+**Severity: Medium**
+
+If a game is reported as maybe valid when it is not, an attacker can prove a withdrawal that is invalid. If
+[aASR-002](#aasr-002-anchorstateregistry-correctly-distinguishes-valid-games) holds, this may not result in severe
+consequences, but would negatively impact system hygiene.
+
+#### Mitigations
+
+- Pending audit of `AnchorStateRegistry`
+- Integration testing
+
+### aASR-002: AnchorStateRegistry correctly distinguishes valid games
+
+We assume that the AnchorStateRegistry correctly reports whether a game is a [**valid
+game**](./anchor-state-registry.md#valid-game).
+
+#### Impact
+
+**Severity: Critical**
+
+If a game is reported as valid when it is not, we may finalize a withdrawal that is invalid. This would result to a loss
+of funds and a loss of confidence in the system.
+
+#### Mitigations
+
+- Pending audit of `AnchorStateRegistry`
+- Integration testing
+
+### aSC-001: SuperchainConfig correctly reports system pause status
+
+We assume SuperchainConfig correctly returns system pause status.
+
+#### Impact
+
+**Severity: Critical**
+
+If SuperchainConfig incorrectly reports system pause status, we may prove / finalize a withdrawal when the system is
+paused. This would create bad system hygiene, and could lead to a loss of funds or a loss of confidence in the system.
+
+#### Mitigations
+
+- Existing audit of `SuperchainConfig`
+
+## Top-Level Invariants
+
```

Comment:
- We assume that the merkle trie proof has bugs with some small non-trivial percent chance
- We must provide a mechanism by which an invalid merkle trie proof can be rejected

Reply:
- A withdrawal transaction must only be finalized if it is valid
-> What does valid mean?
-> Valid means it was truly included in the state of the L2ToL1MessagePasser

- isWithdrawalValid(wd)
-> We rely on the invariant from AnchorStateRegistry that the state is valid
-> We assume that the Merkle Trie library can have bugs
-> Therefore we must provide a mechanism to invalidate bad proofs

^ draft content